src/main/java/KilogramsServlet.java

```diff
@@ -0,0 +1,49 @@
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+@WebServlet("/kilograms")
+public class KilogramsServlet extends HttpServlet {
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        String kg = request.getParameter("kg");
+        String g = request.getParameter("g");
+        String mg = request.getParameter("mg");
+
+        response.setCharacterEncoding("UTF-8");
+        response.setContentType("text/html");
+        PrintWriter writer = response.getWriter();
+
+        int kgToGrams = Integer.valueOf(kg) * 1000;
```

Comment:
Tutaj zaczynasz konwertować elementy które nie sprawdziłaś czy zostały wprowadzone przez użytkownika. Zobacz na początek filmiku z poprzednich zajęć gdzie pokazuję jak to sprawdzić. Generalnie trzeba najpierw sprawdzić czy parametr nie jest `null` oraz czy nie jest pusty. Jeżeli takie sprawdzenie okaże się prawdą to znaczy że użytkownik wpisał ten właśnie parametr - i tutaj zaczynasz dopiero konwersję. Takich ifów musisz mieć 3 dla kilogramów, miligramow oraz gramów. Tyle że uprzenio musisz się upewnić (dokładnie w ten sam sposób) ile parametrów wpisał użytkownik - jeżeli więcej niż 1 to zwracasz mu informację że tak nie może być.

Reply:
poprawione

Reply:
poprawione

Reply:
duże wartości "dziwnie" się wyświetlają np tak "miligramy: 1.2E7" po przeliczeniu z kg, co oznacza ten zapis? notacja?